include/bsp/xi_bsp_fwu.h

```diff
@@ -7,30 +7,118 @@
 #ifndef __XI_BSP_FWU_H__
 #define __XI_BSP_FWU_H__
 
+/**
+ * @file xi_bsp_fwu.h
+ * @brief Xively Client's Board Support Platform (BSP) for Firmware Update
+ *
+ * This file defines the Firmware Update (FWU) API used by the Xively C Client.
+ *
+ * The Xively C Client calls these functions during the Secure File Transfer (SFT) and
+ * Firmware Update (FWU) processes. Some of the functions are event notifications
+ * reflecting the status of the update process, some are device specific action requests
+ * others are checksum generators.
+ */
+
 #include <xively_error.h>
 #include <stdint.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+/**
+ * @function
+ * @brief Decides whether a resource is firmware.
+ *
+ * In Xively's Firmware Update service the application has to be aware of how
+ * a firmware resource is named. If the Xively C Client encounters a resource
+ * which is considered as firmware it will activate the FWU process on it.
+ *
+ * @param [in] resource_name name of the resource the decision has to be based on
+ * @return 1 if the resource is firmware, 0 otherwise
+ */
 uint8_t xi_bsp_fwu_is_this_firmware( const char* const resource_name );
 
+/**
+ * @function
+ * @brief This is an event notification function called when the Xively C Client
+ *        considers the new firmare functional.
+ *
+ * Called by the updated/new firmware.
+ */
 void xi_bsp_fwu_on_new_firmware_ok();
 
+/**
+ * @function
+ * @brief This is an event notification function called when the Xively C Client
+ *        considers the new firmware misfunctional.
+ *
```

Comment:
it would be helpful to list some scenarios when the library might invoke this function, since we don't have it documented anywhere else.

Reply:
I've extended this.